checkfavors.lic

```diff
@@ -9,7 +9,8 @@ class CheckFavors
   def initialize
     @settings = get_settings
     town_data = get_data('town')
-    @hometown = town_data[@settings.hometown]
+    @character_hometown = @settings.favor_town || @settings.fang_cove_override_town || @settings.hometown
```

Comment:
If this is the only use of this variable  (character_hometown) it need not be an instance variable. treat it like town_data. In fact, with a script this small, you can pass all of this, and none of them need be an instance variable.  That would eliminate any chance of interactions between scripts. See recent changes to magic-training for examples.

Reply:
Re: character_hometown, I pulled that trick from another script with a similar use of @hometown and an existing override. I think it was sell-loot.

Reply:
It always works until it doesnt, at which point a lot of people become upset, and suddenly youre scrambling to figure out why. This suggestion is one way to hedge against that.

Reply:
I'm wrong about this. carry on! Well, not about the scrambling, but def the variables.

Reply:
Changed @character_hometown to a local variable.